just-the-code.py

```diff
@@ -0,0 +1,240 @@
+    '''
+    
+    #if type(list_of_PIL_images) != str: 
+       # return 'contact sheet here :)'
+    
+    nrows = int(len(list_of_PIL_images)/5) # set up for 5 (columns) per row
+    contact_sht = Image.new(list_of_PIL_images[0].mode, (sq_size*5, sq_size*(nrows+1)))
+    
+    (x,y) = (0,0)
+    for img in list_of_PIL_images:
+        # paste in an image from the list
+        img.thumbnail((sq_size,sq_size))
+        contact_sht.paste(img, (x, y))
+    
+        # Now we update our coordinate position. If our x-position is going to be at the right-edge of the image, then we set it back to 0 (and update y as well) to point to the next "line" of the contact sheet.
+        if x+sq_size == contact_sht.width:
+            x = 0
+            y = y+sq_size
+        else:
+            x = x+sq_size
+      
+    
+    #contact_sht = contact_sht.resize((int(contact_sht.width/3),int(contact_sht.height/3) ))
+    return contact_sht
+
+#contact_sheet([Image.open('readonly/floyd.jpg'), Image.open('readonly/Noisy_OCR.PNG')])
+
+
+# In[6]:
+
+
+def run(thezipfile=smallFile, theword='Christopher'):
+    '''
+    the overarching logic
+    '''
+    
+    # initialize the dictionary of file entries
+    d = zip_to_dict(thezipfile)
+    
+    for ky in d:
+    #for ky in list(d.keys())[:1]:        # uses only one file for testing
+        PILimage = d[ky]['im']
+        
+        # Look for the word. 
+        word_search_result = bin_read_search( PILimage, theword )  # returns a boolean of the word search's success
+        if not word_search_result: continue
+        
+        # The target word is in the file...
+        print('Results found in file {}'.format(ky))
+        
+        # ... so look for faces.
+        try:
+            faces_coords_list = find_face_boxes(PILimage)
+        except:
+            faces_coords_list = []   # I don't know if this try/except is necessary. What happens with find_face_boxes() when it doesn't find anything??
+        
+        # If there are no faces... make and store that face-search-result-image, and continue to the next file
+        if len(faces_coords_list) < 1:  #there could be a failure here. Again the question: what exactly does find_face_boxes() return?
+            print('But there were no faces in that file!')
+            continue
+        
+        # So there is a face! Crop it/them, and store face-search-result in the dictionary
+        face_pics_list = []
+        for (x1,y1,x2,y2) in faces_coords_list:
+            facePIL = PILimage.crop((x1,y1,x2,y2))
+            face_pics_list.append(facePIL)
+        display(contact_sheet(face_pics_list))
+        
+        
+    #Let's take a look: pretty print (as strings) the dictionary contents 
+    # print(json.dumps( d , indent=4, default=str))
+
+
+# In[7]:
+
+
+run(smallFile, 'Christopher')
+
+
+# In[8]:
+
+
+run(largeFile,'Mark')
+
+
+# In[9]:
+
+
+# goodbye
+smallFile.close()
+largeFile.close()
```

Comment:
This is okay for a script but for more serious coding this can cause problems because if something breaks earlier on these lines won't execute which can be problematic.